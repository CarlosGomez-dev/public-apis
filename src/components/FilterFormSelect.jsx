import { memo } from 'react';

export const FilterFormSelect = ({
  label,
  optionsArray,
  defaultValue = null,
  onChange,
  hasNullOption = false,
}) => (
  <div className='form-select'>
    <label htmlFor={label.replace(' ', '-')} className='label'>
      {label}
    </label>
    <select
      id={label.replace(' ', '-')}
      defaultValue={hasNullOption ? '-' : defaultValue}
      onChange={event => onChange(event.target.value)}
      className='select'
    >
      {hasNullOption ? <option value=''>-</option> : null}
      {optionsArray.map(option => {
        if (Array.isArray(option)) {
          return (
            <option key={option[0]} value={option[0]}>
              {option[1]}
            </option>
          );
        }
        return (
          <option key={option} value={option}>
            {option}
          </option>
        );
      })}
    </select>
  </div>
);

export const FilterFormSelectMemo = memo(FilterFormSelect);
