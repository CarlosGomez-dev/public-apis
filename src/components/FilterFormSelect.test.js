import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { FilterFormSelect } from './FilterFormSelect';

const filterFormSelectProps = {
  label: 'Select label',
  optionsArray: ['2', '4', '6', '8', '10'],
  defaultValue: '4',
  onChange: jest.fn(),
  hasNullOption: false,
};

describe('Filter Form Select', () => {
  it('matches snapshot', () => {
    const { container } = render(<FilterFormSelect {...filterFormSelectProps} />);
    expect(container).toMatchSnapshot();
  });
  it('renders label and default value', () => {
    render(<FilterFormSelect {...filterFormSelectProps} />);
    screen.getByLabelText(filterFormSelectProps.label);
    screen.getByDisplayValue(filterFormSelectProps.defaultValue);
  });
  it('triggers action on option change', () => {
    const { optionsArray } = filterFormSelectProps;
    render(<FilterFormSelect {...filterFormSelectProps} />);
    const select = screen.getByDisplayValue(filterFormSelectProps.defaultValue);
    const options = screen.getAllByRole('option');
    expect(options.length).toBe(optionsArray.length);
    userEvent.selectOptions(select, optionsArray[2]);
    expect(options[2].selected).toBe(true);
  });
});
