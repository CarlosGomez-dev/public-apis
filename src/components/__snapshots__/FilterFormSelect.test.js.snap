// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Form Select matches snapshot 1`] = `
<div>
  <div
    class="form-select"
  >
    <label
      class="label"
      for="Select-label"
    >
      Select label
    </label>
    <select
      class="select"
      id="Select-label"
    >
      <option
        value="2"
      >
        2
      </option>
      <option
        selected=""
        value="4"
      >
        4
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="10"
      >
        10
      </option>
    </select>
  </div>
</div>
`;
